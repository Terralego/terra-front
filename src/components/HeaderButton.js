import React from 'react';
import PropTypes from 'prop-types';

import { Button } from '@blueprintjs/core';

export const HeaderButton = ({ id, icon, alt }) => (
  <Button
    className={`header-button ${id}`}
    minimal
    id={id}
  >
    {icon
      ? <img src={icon} alt={alt} />
      : <span>{alt}</span>}
  </Button>
);

HeaderButton.propTypes = {
  id: PropTypes.string.isRequired,
  icon: PropTypes.string,
  alt: PropTypes.string,
};

HeaderButton.defaultProps = {
  icon: '',
  alt: '',
};

export default HeaderButton;
