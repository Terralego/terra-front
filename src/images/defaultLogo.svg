<svg xmlns="http://www.w3.org/2000/svg" width="36" height="36" fill="#FFFFFF">
    <path d="M35.525 31.228l-8.88-8.882a14.254 14.254 0 002.298-7.752c0-7.909-6.438-14.343-14.346-14.343C6.686.251.254 6.685.254 14.594c0 7.911 6.433 14.344 14.343 14.344 2.856 0 5.513-.849 7.752-2.294l8.88 8.88a.761.761 0 001.076 0l3.22-3.221a.756.756 0 000-1.075zM4.81 14.593c0-5.396 4.391-9.788 9.788-9.788 5.398 0 9.787 4.392 9.787 9.788 0 5.398-4.389 9.789-9.787 9.789-5.398 0-9.788-4.391-9.788-9.789z"/>
</svg>
