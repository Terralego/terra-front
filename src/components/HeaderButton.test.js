import React from 'react';
import renderer from 'react-test-renderer';
import { HeaderButton } from './HeaderButton';

import defaultLogo from '../images/defaultLogo.svg';

it('should render correctly', () => {
  const tree = renderer.create(
    <HeaderButton
      id="theme"
      icon={defaultLogo}
      alt="default logo"
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render correctly without icon', () => {
  const tree = renderer.create(
    <HeaderButton
      id="theme"
      alt="default logo"
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
