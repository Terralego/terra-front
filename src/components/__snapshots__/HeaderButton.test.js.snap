// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<button
  className="bp3-button bp3-minimal header-button theme"
  id="theme"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  type="button"
>
  <span
    className="bp3-button-text"
  >
    <img
      alt="default logo"
      src="defaultLogo.svg"
    />
  </span>
</button>
`;

exports[`should render correctly without icon 1`] = `
<button
  className="bp3-button bp3-minimal header-button theme"
  id="theme"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  type="button"
>
  <span
    className="bp3-button-text"
  >
    <span>
      default logo
    </span>
  </span>
</button>
`;
